main/src/library/Set.flix

```diff
@@ -311,6 +311,12 @@ namespace Set {
         let Set(t) = s;
         RedBlackTree.foldRightWithCont((k, _, c) -> f(k, c), z, t)
 
+    ///
+    /// Returns the result of mapping each element and combining the results.
+    ///
+    pub def foldMap(f: a -> b & ef, s: Set[a]): b & ef with Monoid[b] =
+        foldLeft((acc, x) -> Monoid.combine(f(x), acc), Monoid.empty(), s)
```

Comment:
I just noticed that if I run this line without the `f`
```
def main(): Unit & Impure =
    ("a" :: "b" :: "c" :: Nil) |> List.foldLeft((acc, x) -> Monoid.combine(x, acc), Monoid.empty()) |> println
```

the result is `cba`, whereas I expected it to be `abc` ...

Reply:
I took the foldMap from `Foldable` :
```
pub def foldMap(f: a -> b & ef, t: t[a]): b & ef with Monoid[b] =
    Foldable.foldLeft((acc, x) -> Monoid.combine(f(x), acc), Monoid.empty(), t)
```

Reply:
> I took the foldMap from `Foldable` :
> 
> ```
> pub def foldMap(f: a -> b & ef, t: t[a]): b & ef with Monoid[b] =
>     Foldable.foldLeft((acc, x) -> Monoid.combine(f(x), acc), Monoid.empty(), t)
> ```

So I think acc and f(x) should be swapped here, to preserve the order.

Reply:
I think this also means that foldMap in foldable is wrong and should be fixed.

Reply:
Please add tests that ensure this.

For example, `foldMap(toString, 1 :: 2 :: 3 :: Nil) = "123"` etc